### ISSUE TITLE
Make torch BLAS backend configurable

### ISSUE THREAD

**Issue by dietervdb-meteo** [ecmwf/anemoi-core#1041]
### Is your feature request related to a problem? Please describe.

On ROCm devices the default torch BLAS backend has switched from rocBLAS to hipBLASLt since version 2.7. With this switch also the automatic protection against f16 subnormals being flushed to zero (FTZ) in backward matric multiplications has disappeared. Such FTZ can destabilize training and leads to very different training trajectories depending on the Pytorch version.

<img width="1500" height="750" alt="Image" src="https://github.com/user-attachments/assets/0e13055d-8612-4c95-a379-15ec0f218377" />

See this [report](https://md.sigma2.no/ifpHi-5OS0eO3CnApzu62g) for a more in depth discussion of the issue and its direct link to the BLAS backend and FTZ.

### Describe the solution you'd like

Pytorch allows for overriding the default BLAS backend, one can e.g. force it to use `rocBLAS` over `hipBLASLt` via `torch.backends.cuda.preferred_blas_library("cublas")`. This solves the issue and makes f16 training on ROCm devices continuous along torch versions.

<img width="1500" height="750" alt="Image" src="https://github.com/user-attachments/assets/8bd9fb35-bc58-4bc4-8cc6-f1d713bc8b1b" />

We suggest to have an optional config entry in anemoi that allows to select the blas backend used by Pytorch.

### Describe alternatives you've considered

An alternative is to use `bf16` precision in training, but this comes with issues of its own (see again this [report](https://md.sigma2.no/ifpHi-5OS0eO3CnApzu62g).

### Organisation

RMI

**Comment by dietervdb-meteo** [comment]
Proposed solution has been merged in #1017 . Keeping the issue open for a while to give visibility (on suggestion of @anaprietonem) with the `known-issue` label.

**Comment by dietervdb-meteo** [comment]
From torch > 2.12.0 onwards this issue should be resolved and one should be able to use the default backend again: https://github.com/huydhn/pytorch/commit/256ebd1b2fdde34750a7512169d6eca5845987e6
I hope to explicitly test this in the future once our software stack reaches that.
